medusa/classes.py

```diff
@@ -353,6 +353,10 @@ def select_series(self, all_series):
                             if cur_show not in search_results:
                                 search_results += [cur_show]
 
+                    # add the show if it's a digit search term
+                    if search_term.isdigit() and str(search_term) == str(cur_show['id']):
+                        search_results += [cur_show]
```

Comment:
I think it also needs a check if it already exists in search results (see above).

Reply:
The check is done above. Just below the for:

Reply:
You mean
```
if cur_show not in search_results:
    search_results += [cur_show]
```
?

The problem is if that passes the show can still be added twice.

Reply:
That's what this is for. 
https://github.com/pymedusa/Medusa/blob/2717afdd20f40526aa9aa15faed9a598928575b2/medusa/classes.py#L339